cdc/log.cc

```diff
@@ -635,10 +542,164 @@ class transformer final {
         m.set_cell(ck, _op_col, atomic_cell::make_live(*_op_col.type, ts, _op_col.type->decompose(operation_native_type(op)), _cdc_ttl_opt));
     }
 
-    void set_ttl(const clustering_key& ck, api::timestamp_type ts, gc_clock::duration ttl, mutation& m) const {
-        m.set_cell(ck, _ttl_col, atomic_cell::make_live(*_ttl_col.type, ts, _ttl_col.type->decompose(ttl.count()), _cdc_ttl_opt));
+    void set_ttl(const clustering_key& ck, api::timestamp_type ts, gc_clock::rep ttl, mutation& m) const {
+        m.set_cell(ck, _ttl_col, atomic_cell::make_live(*_ttl_col.type, ts, _ttl_col.type->decompose(ttl), _cdc_ttl_opt));
+    }
+
+    void set_metadata_columns(
+            const clustering_key& delta_ck,
+            const std::optional<clustering_key>& preimage_ck,
+            api::timestamp_type ts,
+            operation op,
+            gc_clock::rep ttl,
+            mutation& log_mut) const {
+        set_operation(delta_ck, ts, op, log_mut);
+        if (ttl) {
+            set_ttl(delta_ck, ts, ttl, log_mut);
+        }
+
+        if (preimage_ck) {
+            set_operation(*preimage_ck, ts, operation::pre_image, log_mut);
+        }
     }
 
+    void set_regular_columns(
+            api::timestamp_type ts,
+            row_update&& row_up, const clustering_key& delta_ck,
+            const cql3::untyped_result_set_row* preimage_row, const std::optional<clustering_key>& preimage_ck,
+            column_kind ckind, mutation& log_mut) const {
+        for (auto& up: row_up.atomic_entries) {
+            auto& cdef = _schema->column_at(ckind, up.id);
+            auto& dst = *_log_schema->get_column_definition(to_bytes("_" + cdef.name()));
```

Comment:
Shouldn't this be done in a central place to ensure consistency?

Reply:
What do you mean?

Reply:
The `"_" + cdef.name()` computation. Isn't this the standard way to get the cdc table column name from the base column one?